index.js

```diff
@@ -99,10 +129,10 @@ function compatibleMessage(thrown, errMatcher) {
 function getConstructorName(errorLike) {
   var constructorName = errorLike;
   if (errorLike instanceof Error) {
-    constructorName = errorLike.constructor.name;
+    constructorName = getFunctionName(errorLike.constructor);
   } else if (typeof errorLike === 'function') {
     // if `err` is not an instance of Error it is an error constructor itself
-    constructorName = new errorLike().name; // eslint-disable-line new-cap
+    constructorName = getFunctionName(errorLike); // eslint-disable-line new-cap
```

Comment:
Line 135: What was the purpose of the `new` in the previous version? Just want to verify it wasn't important, since that aspect is lost in the new version. Also, with the `new` gone, `// eslint-disable-line new-cap` is no longer needed.